src/Microsoft.AspNetCore.JsonPatch/Internal/IAdapterWithTest.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.AspNetCore.JsonPatch.Internal
+{
+    public interface IAdapterWithTest : IAdapter
```

Comment:
Why is this a separate interface? Isn't `IAdapter` internal so we're free to change it? Or did we try to be super careful?

Reply:
Oh good point about that interface. Maybe we can get away with just having one interface on the internal side?